packages/browser-pool/src/browser-pool.ts

```diff
@@ -404,8 +404,16 @@ export class BrowserPool<
 
         // Limiter is necessary - https://github.com/apify/crawlee/issues/1126
         return this.limiter(async () => {
-            let browserController = this._pickBrowserWithFreeCapacity(browserPlugin, { proxyTier });
-            if (!browserController) browserController = await this._launchBrowser(id, { browserPlugin, proxyTier });
+            let browserController;
+            if (proxyTier) {
+                browserController = this._pickBrowserWithFreeCapacity(browserPlugin, { proxyTier });
+            } else if (proxyUrl) {
+                browserController = this._pickBrowserWithFreeCapacity(browserPlugin, { proxyUrl });
+            } else {
+                browserController = this._pickBrowserWithFreeCapacity(browserPlugin);
+            }
```

Comment:
this block feels a bit weird, why dont you remove the branching here, pass both options and handle the priority in the method implementation instead?